packages/core/lib/configureGanacheOptions.js

```diff
@@ -0,0 +1,48 @@
+function sanitizeGanacheOptions(ganacheOptions) {
+  const network_id = ganacheOptions.network_id;
+
+  // Use default network_id if "*" is defined in config
+  if (network_id === "*") {
+    return { ...ganacheOptions, network_id: Date.now() };
+  }
+
+  const parsedNetworkId = parseInt(network_id, 10);
+  if (isNaN(parsedNetworkId)) {
+    const error =
+      `The network id specified in the truffle config ` +
+      `(${network_id}) is not valid. Please properly configure the network id as an integer value.`;
+    throw new Error(error);
+  }
+  return { ...ganacheOptions, network_id: parsedNetworkId };
+}
+
+function configureManagedGanache(config, networkConfig, mnemonic) {
+  const ganacheOptions = {
+    host: networkConfig.host || "127.0.0.1", // Default host for managed ganache
+    port: networkConfig.port || 9545, // Default port for managed ganache
+    network_id: networkConfig.network_id || Date.now(),
+    total_accounts:
+      networkConfig.accounts || networkConfig.total_accounts || 10,
+    default_balance_ether:
+      networkConfig.defaultEtherBalance ||
+      networkConfig.default_balance_ether ||
+      100,
+    blockTime: networkConfig.blockTime || 0,
+    fork: networkConfig.fork,
+    mnemonic: mnemonic,
+    gasPrice: networkConfig.gasPrice || 0x77359400,
+    time: networkConfig.genesis_time || config.genesis_time,
```

Comment:
I think we can probably omit `config.genesis_time`... I guess that's technically breaking, but the whole thing is kind of a bug and I doubt people are actually using it.  Although, IDK, maybe it's best to leave it in just for safety.  (@gnidan, any opinion by any chance?)  I guess probably it's best to leave it in after all, but it might be worth adding a comment noting that this is for compatibility with old bugged versions. :P

Reply:
Also, `time` should be allowed here as a synonym, not just `genesis_time`.  That is something that should actually change.

Reply:
I tried omitting `config.genesis_time` and the CI failed in a few tests which pointed out invalid time if there is no `genesis_time` defined in the config. So, I added that again. Not exactly sure about the behavior though. May be @davidmurdoch can better explain that.

I will add `time` as a synonym here. Thanks.

Reply:
Oh hm.  OK, guess we better not break that then, yikes!